### DISCUSSION TITLE
Add support for shipping package as a Snap

### DISCUSSION THREAD

**Discussion by psadi** [ghostty-org/ghostty#3153] [Feature Requests, Ideas] [answered]
Add support for shipping Ghostty as a [Snap](https://snapcraft.io/) package. This will enable users to easily install and manage on Ubuntu-based systems (which ships with snap by default) or any other distribution with snap installed.

Please let me know if this already planned. I do have some experience in creating snaps and can contribute if its currently not in the pipeline but willing to consider it.

If yes, I request to first [registering-your-app-name](https://snapcraft.io/docs/registering-your-app-name) asap to avoid conflicts in the snapstore.

**Comment by mitchellh** [comment]
Issues are not for feature requests. Converted to a discussion.

**Comment by rhodes-b** [comment]
@psadi on Linux it is up to the community to package ghostty for their respective platforms as listed
https://ghostty.org/docs/install/package
https://github.com/ghostty-org/ghostty/blob/main/PACKAGING.md
https://ghostty.org/docs/install/binary#linux-(official)

**Comment by psadi** [comment]
Yes agreed, It is recommended to first register the snap name by the author and then configuration for the relevant packaging can be contributed accordingly.

Also it seems the name is already registered when cross-checking so my hands are pretty much tied here to initiate and proceed.

```bash
>  snapcraft register ghostty --yes
Store operation failed:
- already_registered: The snap name "ghostty" is already registered.
```

@mitchellh  - Could you confirm if this has been initiated from your end?

**Comment by Footur** [comment]
A ghostty package is now available.

Installation: `snap install --classic --edge ghostty`

But I'm getting an error.

```
snap run ghostty                                                                                           
info: ghostty version=1.0.2-add_snap_package+71606a68                                                        
info: ghostty build optimize=ReleaseFast                                                                     
info: runtime=apprt.Runtime.gtk                                                                              
info: font_backend=font.main.Backend.fontconfig_freetype                                                     
info: dependency harfbuzz=8.4.0                                                                              
info: dependency fontconfig=21402
info: renderer=renderer.OpenGL
info: libxev backend=main.Backend.io_uring
info(os): setlocale from env result=C.UTF-8
info(gtk): GTK version build=4.14.2 runtime=4.14.2
info: reading configuration file path=/home/footur/.config/ghostty/config
info(config): default shell source=env value=/usr/bin/zsh
info(gtk): libadwaita version build=1.5.0 runtime=1.5.0
Gsk-Message: 09:41:52.273: Failed to realize renderer of type 'GskGLRenderer' for surface 'GdkWaylandToplevel
': libEGL not available

Gsk-Message: 09:41:52.273: Failed to realize renderer of type 'GskNglRenderer' for surface 'GdkWaylandTopleve
l': libEGL not available

Gsk-Message: 09:41:52.273: Failed to realize renderer of type 'GskGLRenderer' for surface 'GdkWaylandToplevel
': libEGL not available

Gsk-Message: 09:41:52.273: Failed to realize renderer of type 'GskVulkanRenderer' for surface 'GdkWaylandTopl
evel': Vulkan support disabled via GDK_DEBUG

warning(gtk_wayland): can't set blur: org_kde_kwin_blur_manager protocol unavailable
error(gtk_surface): surface failed to realize: libEGL not available
warning(gtk_surface): this error is usually due to a driver or gtk bug
warning(gtk_surface): this is a common cause of this issue: https://gitlab.gnome.org/GNOME/gtk/-/issues/4950
[1]    108660 segmentation fault (core dumped)  snap run ghostty
```

**Reply by rhodes-b** [reply]
Do you have the proper graphics drivers? You can install them from however your distribution packages mesa

**Reply by Footur** [reply]
`libegl1` and `libegl1-mesa` are installed.

**Comment by mitchellh** [comment, marked as answer]
#3931 is merged. Thanks to @kenvandine. That ensures we have snap builders in CI. Next up is finishing up various release management stuff. I'd love to automate this from our publish release job if possible, @kenvandine if you have Discord please join #packaging and we can talk about it otherwise we can use this forum.
